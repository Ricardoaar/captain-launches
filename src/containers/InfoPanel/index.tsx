import React from "react";

type Props = {
  children: JSX.Element | JSX.Element[];
};

const InfoPanel: React.FC<Props> = ({ children }) => {
  return <section className="info-panel">{children}</section>;
};

export default InfoPanel;
