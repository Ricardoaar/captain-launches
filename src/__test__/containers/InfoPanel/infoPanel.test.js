import { create } from "react-test-renderer";

const { mount } = require("enzyme");
import InfoPanel from "../../../containers/InfoPanel";

describe("InfoPanel", () => {
  const wrapper = mount(
    <InfoPanel>
      <h1>Cats</h1>
    </InfoPanel>
  );
  test("should render correctly", () => {
    expect(wrapper.length).toBe(1);
  });
  test("should render a section", () => {
    expect(wrapper.find("section").length).toEqual(1);
  });
  test("should render children", () => {
    expect(wrapper.find("h1").length).toEqual(1);
  });
  test("should match snapshot", () => {
    const snapWrapper = create(
      <InfoPanel>
        <h1>Cats</h1>
      </InfoPanel>
    );
    expect(snapWrapper.toJSON()).toMatchSnapshot();
  });
});
