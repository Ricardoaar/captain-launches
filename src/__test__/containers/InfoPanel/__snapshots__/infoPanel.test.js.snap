// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel should match snapshot 1`] = `
<section
  className="info-panel"
>
  <h1>
    Cats
  </h1>
</section>
`;
